spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -724,54 +744,110 @@ def write_sorting(sorting, save_path, **nwbfile_kwargs):
+                        unit_id=id) / fs  # spike trains withinin the SortingExtractor object are not scaled by sampling frequency
                     nwbfile.add_unit(id=id, spike_times=spkt)
 
             # Units properties
             for pr in all_properties:
-                unit_ids = [int(k) for k, v in sorting._properties.items()
+                unit_ids = [int(k) for k, v in total_prop_dict.items()
                             if pr in v]
-                vals = [v[pr] for k, v in sorting._properties.items()
-                        if pr in v]
+                all_units = sorting.get_unit_ids()
+                vals = []
+
+                # We can always guarantee two levels of nested dictionaries
+                # based on structure of set_unit_property
+                # Find non null unit property
+                for u in all_units:
+                    if pr in sorting.get_unit_property_names(u):
+                        unit_prop = sorting.get_unit_property(u, pr)
+                        break
+                # unit_prop = next(iter(next(iter(total_prop_dict.values())).values()))
+                print('Unit prop', pr, unit_prop)
+                if type(unit_prop) is dict:
+                    item_names = [x for x in unit_prop.keys()]
+                    # This checks for the ordering as well
+                    if sorted(item_names) == sorted(['name', 'description', 'data']):
+                        names = []
+                        descriptions = []
+                        for u in all_units:
+                            if pr in sorting.get_unit_property_names(u):
+                                names.append(sorting.get_unit_property(u, pr)['name'])
+                                descriptions.append(sorting.get_unit_property(u, pr)['description'])
+                        # names = [v[pr]['name'] for k, v in total_prop_dict.items()]
+                        assert all(elem == names[0] for elem in names), \
+                            'The "name" key in the dictionary of each unit should be the same!'
+                        # descriptions = [v[pr]['description'] for k, v in total_prop_dict.items()]
+                        assert all(elem == descriptions[0] for elem in descriptions), \
+                            'The "description" key in the dictionary of each unit should be the same!'
+
+                        name = names[0]
+                        description = descriptions[0]
+                        for u in all_units:
+                            if pr in sorting.get_unit_property_names(u):
+                                vals.append(sorting.get_unit_property(u, pr)['data'])
+                            else:
+                                vals.append(np.nan)
+                    else:
+                        print('Warning: NwbSortingExtract.write_sorting() requires, if a unit property' +
+                              'is a dictionary, the keys must be "name, description, data" in order!')
```

Comment:
why "in order"?

Reply:
it should not be in order

Reply:
yeah it could be in any order, it's a dictionart